pgxn/neon_walredo/walredoproc.c

```diff
@@ -1047,6 +1053,31 @@ GetPage(StringInfo input_message)
 }
 
 
+static void
+Ping(StringInfo input_message)
+{
+	int			tot_written;
+	/* Response: the input message */
+	tot_written = 0;
+	do {
+		ssize_t		rc;
+		static const char response[BLCKSZ] = {0};
```

Comment:
```suggestion
		static const PGIOAlignedBlock response;
```

Just to avoid any extra overhead caused by bad alignment. Probably doesn't make much difference here. (Real pages coming from the buffer cache are also aligned)

Reply:
PGIOAlignedBlock only exists in pg16. PGAlignedBlock from pg14 and pg15 is only 8-byte-aligned.

I don't think this makes any meaningful difference to performance.

Do you insist on this change? If so, how do I ifdef this correctly so that pg16 gets the same perf?

Reply:
I don't insist, it doesn't matter that much.

The simplest #ifdef would be something like:
```
#if PG_VERSION_NUM >= 160000
static const PGIOAlignedBlock response;
#else
static const PGAlignedBlock response;
#endif
```
Before v16, the buffers in the shared buffer cache were only 8-byte aligned too, so it's fair to not try to do better than that in earlier versions.